sdks/v4-sdk/src/entities/pool.ts

```diff
@@ -160,6 +160,20 @@ export class Pool {
     return this.involvesCurrency(currency)
   }
 
+  /**
+   * v4-only involvesToken convenience method, used for mixed route ETH <-> WETH connection only
+   * @param currency
+   */
+  public v4InvolvesToken(currency: Currency): boolean {
+    return (
+      this.involvesCurrency(currency) ||
+      currency.wrapped.equals(this.currency0) ||
+      currency.wrapped.equals(this.currency1) ||
+      currency.wrapped.equals(this.currency0.wrapped) ||
+      currency.wrapped.equals(this.currency1.wrapped)
```

Comment:
would these last 2 cases ever be hit?
surely any case where `currency.wrapped.equals(this.currency0.wrapped)`, will already have returned `true` in `this.involvesCurrency` because `currency.equals(this.currency0)`?

Or is there a case where it would only be caught by wrapping them both?